api/oc_ri.c

```diff
@@ -1680,7 +1680,7 @@ oc_ri_free_client_cbs_by_mid(uint16_t mid)
     if (!cb->multicast && !cb->discovery && cb->ref_count == 0 &&
         cb->mid == mid) {
       cb->ref_count = 1;
-      notify_client_cb_timeout(cb);
+      notify_client_cb_with_code(cb, OC_REQUEST_TIMEOUT);
```

Comment:
I think this also shouldn't use  the `OC_REQUEST_TIMEOUT` code.

Reply:
fixed